### PR TITLE
fix: limit with offset should return correct results

### LINKED ISSUE #358: Limit with offset return incorrect results
### Describe the bug

We ignore `offset` parameter for local and global limit operators which cause incorrect result in `SQLQueryTestSuite.scala`.

```
[info] - subquery/in-subquery/in-limit.sql *** FAILED *** (2 seconds, 757 milliseconds)
[info]   subquery/in-subquery/in-limit.sql
[info]   Expected "val1[]d	NULL	16	19	17.0	25...", but got "val1[a	16	12	10	15.0	20.0	2000	2014-07-04 01:01:00	2014-07-04
[info]   val1c	8	16	19	17.0	25.0	2600	2014-05-04 01:02:00.001	2014-05-05
[info]   val1]d	NULL	16	19	17.0	25..." Result did not match for query #9
[info]   SELECT *
[info]   FROM   t1
[info]   WHERE  t1c IN (SELECT t2c
[info]                  FROM   t2
[info]                  WHERE  t2b >= 8
[info]                  LIMIT  2
[info]                  OFFSET 2)
[info]   LIMIT 4
[info]   OFFSET 2 (SQLQueryTestSuite.scala:495)
```

Found this when debugging test failure in the PR #250.

### Steps to reproduce

_No response_

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

Closes #358.

### CODE REVIEW

**Comment by viirya** [comment]
cc @sunchao @andygrove

**Review by sunchao — APPROVED** [review]

**Comment by viirya** [comment]
Merged. Thanks.

File: spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala [hunk 1]
```diff
@@ -340,7 +340,7 @@ class CometSparkSessionExtensions
               op
           }
 
-        case op: LocalLimitExec =>
+        case op: LocalLimitExec if getOffset(op) == 0 =>
```
Comment (parthchandra) [inline spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala:343]: Add a `withInfo` for the case when `offset > 0` ?

Comment (viirya) [reply spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala:343]: Oh, I missed it. I will create a follow up. Thanks.

**Pull request merged**